homeassistant/components/amcrest/camera.py

```diff
@@ -92,12 +106,129 @@ def __init__(self, hass, amcrest):
         finally:
             await stream.close()
 
+    # Entity property overrides
+
     @property
     def name(self):
         """Return the name of this camera."""
         return self._name
 
+    @property
+    def device_state_attributes(self):
+        """Return the Amcrest-spectific camera state attributes."""
+        return self._static_attrs
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_ON_OFF
+
+    # Camera property overrides
+
+    @property
+    def is_recording(self):
+        """Return true if the device is recording."""
+        return self._is_recording
+
+    @property
+    def brand(self):
+        """Return the camera brand."""
+        return 'Amcrest'
+
+    @property
+    def model(self):
+        """Return the camera model."""
+        return self._model
+
     @property
     def stream_source(self):
         """Return the source of the stream."""
         return self._camera.rtsp_url(typeno=self._resolution)
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self.video_enabled
+
+    # Additional Amcrest Camera properties
+
+    @property
+    def video_enabled(self):
+        """Return the camera video streaming status."""
+        return self.is_streaming
+
+    @video_enabled.setter
+    def video_enabled(self, enable):
```

Comment:
This isn't appropriate logic for property setter.  Move this to the `turn_on` and `turn_off` methods.

Reply:
Sorry, but I don't understand why you say this (and the corresponding property) is inappropriate. I'm encapsulating the interaction with the device and the manipulation of the corresponding attribute (is_streaming.) Isn't that what properties and setters are for - i.e., encapsulation? Also, this method factors out the logic I would have to duplicate in turn_on and turn_off, which in my programming experience, would be a detrimental change.

Reply:
Properties are expected to be fast, idempotent, safe (no exceptions, except validation) and single principle.  This implementation does may trigger a state update within HA which is far outside of these principles.  If you're concerned about duplication of code, create a private method `_set_video_state(state: bool)`.

Reply:
Ok. Thanks again for the explanation. I'll remove the video_enabled property & setter and move the code from the setter into a private method.